src/modules/facility/facility.service.ts

```diff
@@ -294,20 +301,34 @@ export class FacilityService {
         LenderTypeCode: defaultValues.lenderTypeCode,
       },
       BorrowerParty: {
-        PartyIdentifier: facilityToTranform.obligorPartyIdentifier,
+        PartyIdentifier: facilityToTransform.obligorPartyIdentifier,
       },
       ServicingUser: {
         UserAcbsIdentifier: defaultValues.servicingUser.userAcbsIdentifier,
         UserName: defaultValues.servicingUser.userName,
       },
       CompBalPctReserve: compBalPctReserve,
-      CompBalPctAmount: facilityToTranform.forecastPercentage,
+      CompBalPctAmount: facilityToTransform.forecastPercentage,
       RiskMitigation: {
         RiskMitigationCode: defaultValues.riskMitigationCode,
       },
     };
   }
 
+  private updateAcbsFacilityRequestFieldsForIssueAndAmendExpiryDate(facilityToTransform: BaseFacilityRequestItemWithFacilityIdentifier) {
+    const description = this.buildFacilityDescription(facilityToTransform.productTypeName, facilityToTransform.exposurePeriod);
+    const acbsEffectiveDate = this.dateStringTransformations.addTimeToDateOnlyString(facilityToTransform.effectiveDate);
+    const acbsIssueDate = facilityToTransform.issueDate ? this.dateStringTransformations.addTimeToDateOnlyString(facilityToTransform.issueDate) : null;
+    const { isUserDefinedDate1Zero } = this.buildFacilityStageDerivedValuesToCreate(facilityToTransform.facilityStageCode, facilityToTransform.issueDate);
+    return {
+      Description: description,
+      OfficerRiskDate: acbsEffectiveDate,
+      CreditReviewRiskDate: acbsEffectiveDate,
+      UserDefinedDate1: acbsIssueDate,
```

Comment:
Out of interest does the `buildFacilityStageDerivedValuesToCreate` logic in the base generator already handle `UserDefinedDate1` and `IsUserDefinedDate1Zero` for us how we want it to? If so, we also don't need `UserDefinedDate1` here

Reply:
Oh good point, for `IsUserDefinedDate1Zero` the logic is the same. 
For `UserDefinedDate1` the PUT endpoints set it to `issueDate` from the request body regardless of `facilityStageCode`

Reply:
After discussion with Audrius - the logic for `UserDefinedDate1` in the POST and PUT endpoints should be the same.
When `facilityStageCode === "06"` the facility is unissued and will not have an issueDate, so `UserDefinedDate1` will be null and IsUserDefinedDate1Zero` should be true.
If the facility is issued, the `UserDefinedDate1` should be set to issueDate and IsUserDefinedDate1Zero` should be false.

Reply:
Amazing, thank you very much